x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/generic_confirmation_modal.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactNode, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFormRow,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+interface GenericConfirmationModalProps {
+  description: ReactNode;
+  subheading: ReactNode;
+  target: string;
+  title: ReactNode;
+  onClose(): void;
+  onSave(): void;
+}
+
+export const GenericConfirmationModal: React.FC<GenericConfirmationModalProps> = ({
+  description,
+  onClose,
+  onSave,
+  subheading,
+  target,
+  title,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+
+  const onConfirm = () => {
+    setInputValue('');
+    onSave();
+  };
+
+  return (
+    <EuiModal onClose={onClose}>
```

Comment:
I'd like to take advantage of EuiModal's `initialFocus` prop here to auto-focus the input so users can start typing "DISABLE" immediately if needed!

```suggestion
    <EuiModal onClose={onClose} initialFocus=".euiFieldText">
```

the selector's a little generic but I figured it's the only input in there so no big deal 🤷 we could use data-test-subj if we really wanted but that feels a little odd since it's meant for tests?

Reply:
Actually tacking on another screen reader suggestion here, without the aria-label the screenreader just starts reading out _everything_ in the modal which is a lot

```suggestion
    <EuiModal onClose={onClose} initialFocus=".euiFieldText" aria-label={title}>
```

although we'll have to set the `title` type def to a string and not ReactNode - I think that's probably fine/better though

Reply:
Ohhhh that's interesting. Still a little clunky using this modal with voice over, but that does help.

Reply:
https://github.com/elastic/kibana/pull/83009/commits/e84793010cfaf9ef5d85c54991f3e8ff6c08059e

Reply:
Ya some jank is somewhat unavoidable with modals I think, but aria-labels should help!